PactNet.Tests/PactVerifierConfigTests.cs

```diff
@@ -31,56 +38,43 @@ public void PactVertifier_Init_State_CustomHeadersCountIsZero()
         }
 
         [Fact]
-        [System.Obsolete]
-        public void CustomHeader_backwardCompatibility_ConfirmKeyPairinCustomHeaders()
+        [Obsolete]
+        public void WhenCustomHeaderIsNotNull_ShouldBeAvailableThroughCustomHeadersCollection()
         {
-            string dummy_key = "dummyKey";
-            string dummy_value = "dummyValue";
-
-            _verifierConfig.CustomHeader = new KeyValuePair<string, string>(dummy_key, dummy_value);
+            _verifierConfig.CustomHeader = GetDummyHeader();
 
             Assert.True(_verifierConfig.CustomHeaders.ContainsKey(_verifierConfig.CustomHeader.Value.Key));
         }
 
         [Fact]
-        [System.Obsolete]
-        public void CustomHeader_backwardCompatibility_SetCustomHeaderToNull_ConfirmCustomHeadersCountIsZero()
+        [Obsolete]
+        public void WhenCustomHeaderHasPreviousValue_AndCustomHeaderChangedToNull_ShouldPreviousValueIsNotAvailableInCustomHeaders()
         {
-            string dummy_key = "dummyKey";
-            string dummy_value = "dummyValue";
-
-            _verifierConfig.CustomHeader = new KeyValuePair<string, string>(dummy_key, dummy_value);
+            _verifierConfig.CustomHeader = GetDummyHeader();
 
             _verifierConfig.CustomHeader = null;
 
-            Assert.Equal(0 , _verifierConfig.CustomHeaders.Count);
+            Assert.False( _verifierConfig.CustomHeaders.ContainsKey( GetDummyHeader().Key) );
         }
 
         [Fact]
-        [System.Obsolete]
-        public void CustomHeader_backwardCompatibility_AddCustomHeaderAndConfirmCustomHeadersSizeIsOne()
+        [Obsolete]
+        public void WhenCustomHeaderIsNotNull_CustomHeadersShouldContainKey()
         {
-            string dummy_key = "dummyKey";
-            string dummy_value = "dummyValue";
-
-            _verifierConfig.CustomHeader = new KeyValuePair<string, string>(dummy_key, dummy_value);
+            _verifierConfig.CustomHeader = GetDummyHeader();
 
-            Assert.Equal(1, _verifierConfig.CustomHeaders.Count);
+            Assert.True(_verifierConfig.CustomHeaders.ContainsKey(GetDummyHeader().Key));
```

Comment:
Hmm, 

I am not sure if it was done intentionally, but this test is testing the same thing like test `WhenCustomHeaderIsNotNull_ShouldBeAvailableThroughCustomHeadersCollection`. What I am missing is some test like `WhenCustomHeaderIsNull_ShouldNotBeAvailableThroughCustomHeadersCollection `